File: app-serverless/src/main/java/com/xsolla/android/storesdkexample/fragments/InventoryFragment.java
package com.xsolla.android.storesdkexample.fragments;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;

import androidx.appcompat.widget.Toolbar;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.xsolla.android.serverlessexample.R;
import com.xsolla.android.storesdkexample.adapter.InventoryAdapter;
import com.xsolla.android.storesdkexample.data.store.Store;
import com.xsolla.android.storesdkexample.fragments.base.BaseFragment;
import com.xsolla.android.storesdkexample.listener.ConsumeListener;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class InventoryFragment extends BaseFragment implements ConsumeListener {

    private BroadcastReceiver viUpdateReceiver = new BroadcastReceiver() {
        @Override
        public void onReceive(Context context, Intent intent) {
            getItems();
        }
    };

    private InventoryAdapter inventoryAdapter;
    private RecyclerView recyclerView;

    @Override
    public int getLayout() {
        return R.layout.fragment_shop;
    }

    @Override
    public void initUI() {
        recyclerView = rootView.findViewById(R.id.items_rv);
        LinearLayoutManager layoutManager = new LinearLayoutManager(getContext());
        DividerItemDecoration dividerItemDecoration = new DividerItemDecoration(recyclerView.getContext(),
                layoutManager.getOrientation());

        recyclerView.setLayoutManager(layoutManager);
        recyclerView.addItemDecoration(dividerItemDecoration);

        setupToolbar();
    }

    @Override
    public void onResume() {
        super.onResume();
        getItems();
        LocalBroadcastManager.getInstance(requireContext())
                .registerReceiver(viUpdateReceiver, new IntentFilter(Store.ACTION_VI_UPDATE));
    }

    @Override
    public void onPause() {
        super.onPause();
        LocalBroadcastManager.getInstance(requireContext())
                .unregisterReceiver(viUpdateReceiver);
    }

    private void getItems() {
        Store.getInventory(new Store.InventoryCallback() {
            @Override
            public void onSuccess(@NotNull List<Store.InventoryItem> inventoryItems) {
                inventoryAdapter = new InventoryAdapter(inventoryItems, InventoryFragment.this);
                recyclerView.setAdapter(inventoryAdapter);
            }

            @Override
            public void onFailure(@NotNull String errorMessage) {
                showSnack(errorMessage);
            }
        });
    }

    @Override
    public void onSuccess() {
        showSnack("Item consumed");
    }

    @Override
    public void onFailure(String errorMessage) {
        showSnack(errorMessage);
    }

    private void setupToolbar() {
        Toolbar toolbar = rootView.findViewById(R.id.toolbar);
        toolbar.setTitle("Inventory");
        toolbar.setNavigationIcon(R.drawable.ic_arrow_back_white_24dp);
        toolbar.setNavigationOnClickListener(v -> popFragment());
    }

}

File: app-customauth/src/main/java/com/xsolla/android/customauth/ui/store/InventoryFragment.kt
package com.xsolla.android.customauth.ui.store

import androidx.core.content.ContextCompat
import androidx.fragment.app.viewModels
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.DividerItemDecoration
import by.kirich1409.viewbindingdelegate.viewBinding
import com.xsolla.android.customauth.R
import com.xsolla.android.customauth.databinding.FragmentInventoryBinding
import com.xsolla.android.customauth.ui.BaseFragment
import com.xsolla.android.customauth.ui.adapter.ConsumeListener
import com.xsolla.android.customauth.ui.adapter.InventoryAdapter
import com.xsolla.android.customauth.viewmodels.VmInventory
import com.xsolla.android.inventory.XInventory
import com.xsolla.android.inventory.callback.ConsumeItemCallback
import com.xsolla.android.inventory.callback.GetInventoryCallback
import com.xsolla.android.inventory.entity.response.InventoryResponse

class InventoryFragment : BaseFragment(), ConsumeListener {
    private val binding: FragmentInventoryBinding by viewBinding()
    private val viewModel: VmInventory by viewModels()

    private lateinit var adapter: InventoryAdapter

    override fun getLayout() = R.layout.fragment_inventory

    override fun initUI() {
        adapter = InventoryAdapter(listOf(), this)
        binding.recycler.adapter = adapter
        binding.recycler.addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL).apply {
            ContextCompat.getDrawable(requireContext(), R.drawable.item_divider)?.let { setDrawable(it) }
        })
        binding.goToStoreButton.setOnClickListener { findNavController().navigate(R.id.nav_vi) }

        viewModel.inventory.observe(viewLifecycleOwner) { adapter.submitList(it) }
        viewModel.subscriptions.observe(viewLifecycleOwner) { adapter.setSubscriptions(it) }
        viewModel.getItems { showSnack(it) }
        viewModel.getSubscriptions { showSnack(it) }
    }

    override fun onConsume(item: InventoryResponse.Item) {
        consume(item)
    }

    override fun onSuccess() {
        showSnack("Item Consumed")
    }

    override fun onFailure(errorMessage: String) {
        showSnack(errorMessage)
    }

    private fun consume(item: InventoryResponse.Item) {
        XInventory.consumeItem(item.sku!!, 1, null, object : ConsumeItemCallback {
            override fun onSuccess() {
                XInventory.getInventory(object : GetInventoryCallback {
                    override fun onSuccess(data: InventoryResponse) {
                        val items = data.items.filter { item -> item.type == InventoryResponse.Item.Type.VIRTUAL_GOOD }
                        adapter.submitList(items.toList())
                        showSnack("Item consumed")
                    }

                    override fun onError(throwable: Throwable?, errorMessage: String?) {
                        showSnack(errorMessage ?: throwable?.javaClass?.name ?: "Error")
                    }
                })
            }

            override fun onError(throwable: Throwable?, errorMessage: String?) {
                showSnack(errorMessage ?: throwable?.javaClass?.name ?: "Error")
            }
        })
    }
}